modules/apps/commerce/commerce-test/src/testIntegration/java/com/liferay/commerce/service/test/CommerceOrderItemLocalServiceTest.java

```diff
@@ -983,6 +985,68 @@ public void testDeleteProductBundleWithOptionWithSKU() throws Exception {
 		}
 	}
 
+	@Test
+	public void testImportCommerceOrderItem() throws Exception {
+		CPInstance cpInstance = CPTestUtil.addCPInstance(_group.getGroupId());
+
+		_cpInstances.add(cpInstance);
+
+		CPDefinition cpDefinition = cpInstance.getCPDefinition();
+
+		CPOption cpOption = CPOptionLocalServiceUtil.addCPOption(
+			null, _user.getUserId(),
+			HashMapBuilder.put(
+				LocaleUtil.getSiteDefault(), "PCB Reference"
+			).build(),
+			RandomTestUtil.randomLocaleStringMap(), "text", false, false, false,
+			"pcb-reference", _serviceContext);
+
+		_cpOptions.add(cpOption);
+
+		CPDefinitionOptionRel cpDefinitionOptionRel =
+			CPTestUtil.addCPDefinitionOptionRel(
+				_commerceCatalog.getGroupId(), cpDefinition.getCPDefinitionId(),
+				cpOption.getCPOptionId());
+
+		_cpDefinitionOptionRels.add(cpDefinitionOptionRel);
+
+		_commerceInventoryWarehouse =
+			CommerceInventoryTestUtil.addCommerceInventoryWarehouse(
+				ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		_commerceInventoryWarehouseItems.add(
+			CommerceInventoryTestUtil.addCommerceInventoryWarehouseItem(
+				_user.getUserId(), _commerceInventoryWarehouse,
+				BigDecimal.valueOf(2), cpInstance.getSku(), StringPool.BLANK));
+
+		CommerceTestUtil.addWarehouseCommerceChannelRel(
+			_commerceInventoryWarehouse.getCommerceInventoryWarehouseId(),
+			_commerceChannel.getCommerceChannelId());
+
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.addCommerceOrder(
+				_user.getUserId(), _commerceChannel.getGroupId(),
+				_accountEntry.getAccountEntryId(), _commerceCurrency.getCode(),
+				0);
+
+		_commerceOrders.add(commerceOrder);
+
+		String json = StringBundler.concat(
+			"[{\"key\":\"pcb-reference\",\"skuOptionKey\":\"pcb-reference\",",
+			"\"skuOptionName\":\"PCB Reference\",\"value\":[\"PCB XYZ\"],",
+			"\"skuId\":null}]");
+
+		CommerceOrderItem commerceOrderItem =
+			_commerceOrderItemLocalService.importCommerceOrderItem(
+				_user.getUserId(), null, 0, commerceOrder.getCommerceOrderId(),
+				cpInstance.getCPInstanceId(), null, json, BigDecimal.ONE,
+				BigDecimal.ONE, null, null, _serviceContext);
+
+		_commerceOrderItems.add(commerceOrderItem);
+
+		Assert.assertEquals(json, commerceOrderItem.getJson());
+	}
+
```

Comment:
I also don't get why we are adding all the created things in the lists as in the tearDown I think we loop just on the orderItems 😄